Inconsistent report of CS8156 when using generically constrained interfaces and [UnscopedRef]

**Version Used**: 
9.0.100-preview.7.24407.12

**Steps to Reproduce**:


```csharp

[InlineArray(256)]
public struct BufferStruct : IBufferInterface
{
    private byte _data;

    [UnscopedRef]
    public ReadOnlySpan<byte> Data => this;
}


interface IBufferInterface
{
    [UnscopedRef] //<--commenting this out removes the CS8156 error below
    public ReadOnlySpan<byte> Data { get; }
}

struct Foo<T> 
    where T : struct, IBufferInterface
{
    T genericBuffer;
    BufferStruct directBuffer;
    IBufferInterface interfaceBuffer;
    

    //[UnscopedRef] //<-- Has no effect
    public ReadOnlySpan<byte> GetGenericBuffer()
    {
        //error CS8156: An expression cannot be used in this 
        //context because it may not be passed or returned by reference
        return genericBuffer.Data;
    }

    [UnscopedRef]//this is required to fix CS8156
    public ReadOnlySpan<byte> GetDirectBuffer()
    {
        //was an error, but it's fixed by [UnscopedRef]
        return directBuffer.Data;
    }

    public ReadOnlySpan<byte> GetInterfaceBuffer()
    {
        return interfaceBuffer.Data;
    }
}
```


**Diagnostic Id**:
CS8156

**Expected Behavior**:
The call to `genericBuffer` to be identical in behavior to the call to `directBuffer`

**Actual Behavior**:
The call to `genericBuffer` reports error CS8156

I am not sure what I am missing, but I am not observing the reported behavior:
1) CS8156 is not reported, CS8170 is reported insted
2) For usages of ```genericBuffer``` and ```directBuffer``` it is reported or not reported consistently

Here is the code I used:
```
using System;
using System.Diagnostics.CodeAnalysis;
using System.Runtime.CompilerServices;

#pragma warning disable CS0649

[InlineArray(256)]
public struct BufferStruct : IBufferInterface
{
    private byte _data;

    [UnscopedRef]
    public ReadOnlySpan<byte> Data => this;
}


interface IBufferInterface
{
    [UnscopedRef]
    public ReadOnlySpan<byte> Data { get; }
}

struct TestStruct<T> 
    where T : struct, IBufferInterface
{
    T genericBuffer;
    BufferStruct directBuffer;
    IBufferInterface interfaceBuffer;
    
    [UnscopedRef]
    public ReadOnlySpan<byte> GetGenericBuffer1()
    {
        return genericBuffer.Data;
    }

    [UnscopedRef]
    public ReadOnlySpan<byte> GetDirectBuffer1()
    {
        return directBuffer.Data;
    }

    public ReadOnlySpan<byte> GetInterfaceBuffer()
    {
        return interfaceBuffer.Data;
    }
    
    public ReadOnlySpan<byte> GetGenericBuffer2()
    {
#line 1000
        return genericBuffer.Data;
    }

    public ReadOnlySpan<byte> GetDirectBuffer2()
    {
#line 2000
        return directBuffer.Data;
    }
}
```

And the errors I observed:
```
(1000,16,1000,29): error CS8170: Struct members cannot return 'this' or other instance members by reference
(2000,16,2000,28): error CS8170: Struct members cannot return 'this' or other instance members by reference
```

--- a/src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs
+++ b/src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs
@@ -28916,5 +28916,81 @@ interface UsePia5 : ITest29
 
             CompileAndVerify(compilation2, symbolValidator: metadataValidator, verify: Verification.Skipped).VerifyDiagnostics();
         }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/74785")]
+        public void Issue74785()
+        {
+            var src = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+#pragma warning disable CS0649
+
+[InlineArray(256)]
+public struct BufferStruct : IBufferInterface
+{
+    private byte _data;
+
+    [UnscopedRef]
+    public ReadOnlySpan<byte> Data => this;
+}
+
+
+interface IBufferInterface
+{
+    [UnscopedRef]
+    public ReadOnlySpan<byte> Data { get; }
+}
+
+struct TestStruct<T> 
+    where T : struct, IBufferInterface
+{
+    T genericBuffer;
+    BufferStruct directBuffer;
+    IBufferInterface interfaceBuffer;
+    
+    [UnscopedRef]
+    public ReadOnlySpan<byte> GetGenericBuffer1()
+    {
+        return genericBuffer.Data;
+    }
+
+    [UnscopedRef]
+    public ReadOnlySpan<byte> GetDirectBuffer1()
+    {
+        return directBuffer.Data;
+    }
+
+    public ReadOnlySpan<byte> GetInterfaceBuffer()
+    {
+        return interfaceBuffer.Data;
+    }
+    
+    public ReadOnlySpan<byte> GetGenericBuffer2()
+    {
+#line 1000
+        return genericBuffer.Data;
+    }
+
+    public ReadOnlySpan<byte> GetDirectBuffer2()
+    {
+#line 2000
+        return directBuffer.Data;
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net90);
+
+            comp.VerifyEmitDiagnostics(
+                // (1000,16): error CS8170: Struct members cannot return 'this' or other instance members by reference
+                //         return genericBuffer.Data;
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "genericBuffer").WithLocation(1000, 16),
+                // (2000,16): error CS8170: Struct members cannot return 'this' or other instance members by reference
+                //         return directBuffer.Data;
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "directBuffer").WithLocation(2000, 16)
+                );
+        }
     }
 }

